File: connection-backend/src/main/java/de/florianschmitt/service/authorization/UserDetailsServiceImpl.java
package de.florianschmitt.service.authorization;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import de.florianschmitt.model.entities.ESystemUser;
import de.florianschmitt.repository.SystemUserRepository;

@Service
public class UserDetailsServiceImpl implements UserDetailsService {

    @Autowired
    private SystemUserRepository systemUserRepository;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        ESystemUser user = systemUserRepository.findByEmail(username)//
                .orElseThrow(() -> new UsernameNotFoundException("user not found"));
        return user;
    }
}

File: src/main/kotlin/de/florianschmitt/service/authorization/UserDetailsServiceImpl.kt
package de.florianschmitt.service.authorization

import de.florianschmitt.repository.SystemUserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserDetailsServiceImpl : UserDetailsService {

    @Autowired
    private lateinit var systemUserRepository: SystemUserRepository

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String) = systemUserRepository.findByEmail(username)
            .orElseThrow { UsernameNotFoundException("user not found") }
}
